src/vm/generics.cpp

```diff
@@ -240,11 +239,11 @@ ClassLoader::CreateTypeHandleForNonCanonicalGenericInstantiation(
     // Collectible types have some special restrictions
     if (pAllocator->IsCollectible())
     {
-        if (fHasThreadStatics || fHasContextStatics)
+        if (fHasContextStatics)
```

Comment:
You can delete `fHasContextStatics` while you are on it. It is always false in CoreCLR.

Reply:
I have a separate change ready that removes everything around context statics. I didn't want to mix it with this change.